x-pack/plugins/lens/public/datatable_visualization/visualization.tsx

```diff
@@ -383,7 +386,7 @@ export const getDatatableVisualization = ({
                         !datasource!.getOperationForColumnId(column.columnId)?.isBucketed,
                       ],
                       alignment: typeof column.alignment === 'undefined' ? [] : [column.alignment],
-                      colorMode: [column.colorMode ?? 'none'],
+                      colorMode: !canColor ? ['none'] : [column.colorMode ?? 'none'],
```

Comment:
nit, I find it a bit easier this way: 
```suggestion
                      colorMode: [canColor && column.colorMode ? column.colorMode: ‘none’]
```